external/flume/src/test/scala/org/apache/spark/streaming/flume/FlumeStreamSuite.scala

```diff
@@ -76,7 +75,7 @@ class FlumeStreamSuite extends FunSuite with BeforeAndAfter with Matchers with L
 
   /** Find a free port */
   private def findFreePort(): Int = {
-    Utils.startServiceOnPort(23456, (trialPort: Int) => {
+    Utils.startServiceOnPort((math.random * Int.MaxValue).toInt, (trialPort: Int) => {
```

Comment:
If anything, it should be changed to reject an invalid `startPort`, and this code should not pass invalid port values in any event.